### ISSUE TITLE
Office document with lock

### ISSUE THREAD

**Issue by RoyalMist** [OpenMarshal/npm-WebDAV-Server#40]
Hello.

Got an issue for editing word docs through webdav thanks to those kind of url: 
ms-word:ofe|u|http://localhost:1900/files/caf/1563594/rc/1537278137926/Document.docx

I can open it but cannot save changes due to the lock despite the fact I did not have any other user on the file.
![image](https://user-images.githubusercontent.com/709371/45696017-e9e3a700-bb62-11e8-9f84-a0866b9dafd9.png)

Here you can find my server code: 

```
const webdav = require("webdav-server").v2;
const config = require("../configuration/config");
const userManager = new webdav.SimpleUserManager();

let HTTPNoAuthentication = (function() {
  function HTTPNoAuthentication(userManager, realm) {
    if (realm === void 0) {
      realm = "realm";
    }

    this.userManager = userManager;
    this.realm = realm;
  }

  HTTPNoAuthentication.prototype.askForAuthentication = function() {
    return {
      "WWW-Authenticate": `Basic realm="${this.realm}"`
    };
  };

  HTTPNoAuthentication.prototype.getUser = function(ctx, callback) {
    this.userManager.getDefaultUser(function(defaultUser) {
      callback(null, defaultUser);
    });
  };

  return HTTPNoAuthentication;
})();

function setHeaders(arg) {
  if (arg.request.method === "OPTIONS") {
    arg.response.setHeader(
      "Access-Control-Allow-Methods",
      "PROPPATCH,PROPFIND,OPTIONS,DELETE,UNLOCK,COPY,LOCK,MKCOL,MOVE,HEAD,POST,PUT,GET"
    );
    
    arg.response.setHeader("allow", "PROPPATCH,PROPFIND,OPTIONS,DELETE,UNLOCK,COPY,LOCK,MKCOL,MOVE,HEAD,POST,PUT,GET");
    arg.response.setHeader("Access-Control-Allow-Headers", "*");
    arg.response.setHeader("Access-Control-Allow-Origin", "*");
  }

  arg.response.setHeader("MS-Author-Via", "DAV");
}

const server = new webdav.WebDAVServer({
  port: 1900,
  httpAuthentication: new HTTPNoAuthentication(userManager, "Default realm")
});

server.beforeRequest((arg, next) => {
  setHeaders(arg);
  next();
});

server.setFileSystem(config.WEB_DAV_FOLDER, new webdav.PhysicalFileSystem(config.ROOT_FOLDER), success => {
  server.start(() => console.log("READY"));
});
```

Thanks a lot for your help.

**Comment by fabianinnova** [comment]
@RoyalMist , @AdrienCastex I went through it in previous versions, but in the current version this error does not happen. I think maybe it was a bug, but after several tests in ms-office I consider it solved.

**Comment by AdrienCastex** [comment]
Hello,

The error occurred for me. It is a problem with the LOCK method and with the If header.

The fix will be available in the version `2.6.1`.
Thank you very much for your contribution.
